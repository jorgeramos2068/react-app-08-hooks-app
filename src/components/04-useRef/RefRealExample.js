import React, {useState, useRef} from 'react';
import MultipleCustomHooks from '../03-examples/MultipleCustomHooks';
import '../02-useEffect/effects.css';

const RefRealExample = () => {
  const [show, setShow] = useState(false);

  return (
    <div>
      <h1>RefRealExample</h1>
      <hr />
      <button
        className="btn btn-primary mt-3"
        onClick={() => {
          setShow(!show);
        }}
      >
        Show/Hide
      </button>
      { show && <MultipleCustomHooks /> }
    </div>
  );
};

export default RefRealExample;
