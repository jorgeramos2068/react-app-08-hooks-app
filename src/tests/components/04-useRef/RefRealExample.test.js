import React from 'react';
import {shallow} from 'enzyme';
import RefRealExample from '../../../components/04-useRef/RefRealExample';

describe('Tests in <RefRealExample />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<RefRealExample />);
  });

  test('Should display <RefRealExample /> correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Should display <MultipleCustomHooks /> component', () => {
    const button = wrapper.find('button');
    button.simulate('click');
    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
  });
});
