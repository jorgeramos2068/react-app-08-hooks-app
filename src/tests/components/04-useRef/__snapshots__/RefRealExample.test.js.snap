// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <RefRealExample /> Should display <RefRealExample /> correctly 1`] = `
<div>
  <h1>
    RefRealExample
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-3"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
